backend/src/main/java/com/example/dontwastetomuch/game/GameController.java

```diff
@@ -72,10 +71,10 @@ public ResponseEntity<UserGameDTO> getOneOfMyGames(@PathVariable String gameId,
     public ResponseEntity<Void> switchGameStatus(@PathVariable String gameId, Principal principal){
         try {
             MyUser user = myUserRepo.findById(principal.getName()).orElseThrow();
-            ResponseEntity<UserGameDTO> userGameDTO = getOneOfMyGames(gameId, principal);
+            ResponseEntity<Game> game = getOneGameToEdit(gameId, principal);
             Game game1 = new Game();
-            game1.setGameName(Objects.requireNonNull(userGameDTO.getBody()).getGameName());
-            game1.setApproved(userGameDTO.getBody().isApproved());
+            game1.setGameName(Objects.requireNonNull(game.getBody()).getGameName());
+            game1.setApproved(game.getBody().isApproved());
             game1.setId(gameId);
             gameService.switchStatus(game1, user);
```

Comment:
typically not something you would want in controller. seems like a lot of logic so maybe in service for this block?
Also I think `orElseThrow()` throws a NoSuchElement Exception. You only catch IllegalArgumentExceptions so right now if no user is found you would get an internal server error. Your method getOneGameToEdit also throws NSE Exceptions so same problem.
One onther thing, your service does not need to know your principal. So in getOneGameToEdit maybe put principal.getName() as parameter to only retrieve the username.

Reply:
That's true, changed it all, thx again 👍